### ISSUE TITLE
media_characteritics table not included in "full" query

### ISSUE THREAD

**Issue by mattreusswig** [pepijn-devries/ECOTOXr#2]
Hello - 
When performing a query that specifies the "full" set of output fields included in list_ecotox_fields("full"), it appears the media_characteristics table is not being pulled in correctly. Rather than capturing record data, the query outputs the field names for that table in each record.  The reprex below was made using the 12-15-2021 version of ECOTOX.

``` r
library(ECOTOXr)
#> Loading required package: RSQLite
#> ECOTOX database file located, you are ready to go!
  
search <-
  list(
    latin_name    = list(
      terms          = c("Skeletonema", "Daphnia"),
      method         = "contains"
    ),
    chemical_name = list(
      terms          = "Cadmium",
      method         = "exact"
    )
  )

result <- search_ecotox(search, 
                        output_fields = list_ecotox_fields("full"))
```

<sup>Created on 2022-02-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value
#>  version  R version 4.1.2 (2021-11-01)
#>  os       Windows 10 x64 (build 19044)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  English_United States.1252
#>  ctype    English_United States.1252
#>  tz       America/New_York
#>  date     2022-02-02
#>  pandoc   2.14.0.3 @ C:/Program Files/RStudio/bin/pandoc/ (via rmarkdown)
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version date (UTC) lib source
#>  bit           4.0.4   2020-08-04 [1] CRAN (R 4.1.1)
#>  bit64         4.0.5   2020-08-30 [1] CRAN (R 4.1.1)
#>  blob          1.2.2   2021-07-23 [1] CRAN (R 4.1.1)
#>  cachem        1.0.6   2021-08-19 [1] CRAN (R 4.1.2)
#>  cli           3.1.0   2021-10-27 [1] CRAN (R 4.1.1)
#>  crayon        1.4.2   2021-10-29 [1] CRAN (R 4.1.1)
#>  DBI           1.1.1   2021-01-15 [1] CRAN (R 4.1.2)
#>  digest        0.6.28  2021-09-23 [1] CRAN (R 4.1.1)
#>  ECOTOXr     * 0.1.1   2021-10-05 [1] CRAN (R 4.1.2)
#>  evaluate      0.14    2019-05-28 [1] CRAN (R 4.1.1)
#>  fastmap       1.1.0   2021-01-25 [1] CRAN (R 4.1.1)
#>  fs            1.5.0   2020-07-31 [1] CRAN (R 4.1.1)
#>  glue          1.5.0   2021-11-07 [1] CRAN (R 4.1.2)
#>  highr         0.9     2021-04-16 [1] CRAN (R 4.1.1)
#>  htmltools     0.5.2   2021-08-25 [1] CRAN (R 4.1.1)
#>  knitr         1.37    2021-12-16 [1] CRAN (R 4.1.2)
#>  magrittr      2.0.1   2020-11-17 [1] CRAN (R 4.1.2)
#>  memoise       2.0.1   2021-11-26 [1] CRAN (R 4.1.2)
#>  pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 4.1.1)
#>  rappdirs      0.3.3   2021-01-31 [1] CRAN (R 4.1.1)
#>  Rcpp          1.0.8   2022-01-13 [1] CRAN (R 4.1.2)
#>  reprex        2.0.1   2021-08-05 [1] CRAN (R 4.1.1)
#>  rlang         1.0.0   2022-01-26 [1] CRAN (R 4.1.2)
#>  rmarkdown     2.11    2021-09-14 [1] CRAN (R 4.1.1)
#>  RSQLite     * 2.2.8   2021-08-21 [1] CRAN (R 4.1.2)
#>  rstudioapi    0.13    2020-11-12 [1] CRAN (R 4.1.1)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.1.2)
#>  stringi       1.7.5   2021-10-04 [1] CRAN (R 4.1.1)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.1.1)
#>  vctrs         0.3.8   2021-04-29 [1] CRAN (R 4.1.1)
#>  withr         2.4.2   2021-04-18 [1] CRAN (R 4.1.1)
#>  xfun          0.28    2021-11-04 [1] CRAN (R 4.1.2)
#>  yaml          2.2.2   2022-01-25 [1] CRAN (R 4.1.2)
#> 
#>  [1] C:/Users/matt.reusswig/msr_documents/R-4.1.2/library
#> 
#> ------------------------------------------------------------------------------
```

</details>

**Comment by pepijn-devries** [comment]
Hi Matthew,

Thank you for using the ECOTOXr package and taking the time for providing feedback. I'm aware that the current search functionality of the package is limited. I will have a look at your specific issue somewhere in the coming week and will keep you posted.

**Comment by pepijn-devries** [comment]
Hi Matthew,

Sorry for the late response. I'm working on this project volentary in my spare time (which I have to prioritize). As I suspected the "full" list of output fields could not be handled by the simple query builder. Therefore, I've restricted the "full" list of output fields to an extent where it would work. Unfortunately, this also includes the media_characteristics fields. As a workaround you could first perform a search query, then perform a custom query on the media_characteristics selecting only the relevant result_ids.

I have plans to include several custom search queries that would be more efficient than the ones that are now generated automatically. But I can't tell when I will be able to add those to the package.

The changes are now available in the work-in-progress branch. I will merge it into the main branch in a later stage when I also submit to CRAN...

Cheers,

Pepijn

**Comment by pepijn-devries** [comment]
> Hi @pepijn-devries,
> 
> That makes sense. And thanks for publishing your package--its been very helpful and I look forward to see what you do with it in the future!
> 
> Best, Matthew

Sorry for the delay. It took me longer than expected. But this issue should now be fixed in the new release (0.2.0) on CRAN

**Comment by mattreusswig** [comment]
Thanks, @pepijn-devries!
